asf_heat_pump_suitability/getters/get_target.py

```diff
@@ -0,0 +1,183 @@
+
+
+def get_df_target_tenure(censored_values: int = 0) -> pl.DataFrame:
+    """
+    Get dataframe of tenure type counts for all LSOAs in England and Wales, and fill censored values with given
+    constant. Source: census data 2021.
+
+    Args:
+        censored_values (int): value to fill censored values with, 0-9. Default 0.
+
+    Returns:
+        pl.Dataframe: counts of tenure type for all LSOAs in England and Wales
+    """
+    content = base_getters.get_content_from_path(
+        config["data_source"]["EW_census_housing_characteristics"]
+    )
+    df = pl.read_excel(content, sheet_name="3c", engine="calamine")
+
+    # Remove empty header rows
+    df = df.rename(df[2].to_dicts().pop()).slice(
+        3,
+    )
+    df = df.rename(
+        {
+            "Area Code": "lsoa",
+            "Owned or shared ownership": "owner-occupied",
+            "Social Rented": "rental (social)",
+            "Private Rented or lives rent free": "rental (private)",
+        }
+    ).drop(["Area Name"])
+
+    return _fill_df_censored_values(df, censored_values)
+
+
+def get_df_target_tenure_uncensored() -> pl.DataFrame:
+    """
+    Get dataframe of tenure type counts for all LSOAs in England and Wales. Source: census data 2021.
+
+    Returns:
+        pl.Dataframe: counts of tenure type for all LSOAs in England and Wales
+    """
+    df = pl.read_csv(config["data_source"]["EW_census_tenure"])
+
+    owned_cols = [
+        "Owned: Owns with a mortgage or loan",
+        "Owned: Owns outright",
+        "Shared ownership: Shared ownership",
+    ]
+
+    social_rent_cols = [
+        "Social rented: Other social rented",
+        "Social rented: Rents from council or Local Authority",
+    ]
+
+    private_rent_cols = [
+        "Lives rent free",
+        "Private rented: Private landlord or letting agency",
+        "Private rented: Other private rented",
+    ]
+
+    df = (
+        df.rename(
+            {
+                "Lower layer Super Output Areas Code": "lsoa",
+                "Tenure of household (9 categories)": "tenure",
+            }
+        )
+        .pivot(index="lsoa", columns="tenure", values="Observation")
+        .with_columns(
+            [
+                pl.sum_horizontal(owned_cols).alias("owner-occupied"),
+                pl.sum_horizontal(social_rent_cols).alias("rental (social)"),
+                pl.sum_horizontal(private_rent_cols).alias("rental (private)"),
+            ]
+        )
+        .select(
+            pl.col(["lsoa", "owner-occupied", "rental (social)", "rental (private)"])
+        )
+    )
+
+    return df
+
+
+def get_df_target_build_year(
+    year_label: int = 1930,
+    pre_cols: list = config["mapping"]["build_year_pre_cols"],
+    post_cols: list = config["mapping"]["build_year_post_cols"],
+) -> pl.DataFrame:
+    """
+    Get dataframe of counts of properties built before and after given year for all LSOAs in England and Wales. Source:
```

Comment:
Probably need a bit more info in this doc string about the range of years that can be used. I would potentially think I would be able to use any year here but that's not the case right?

Reply:
have updated docstring - see what you think. The split year in this pipeline will depend on the age bands in both target and sample datasets but this specific function will work with any age bands available in this target dataset - so yes, not any year.